python/ray/dag/compiled_dag_node.py

```diff
@@ -538,6 +543,10 @@ def __init__(
                 executions is beyond the DAG capacity, the new execution would
                 be blocked in the first place; therefore, this limit is only
                 enforced when it is smaller than the DAG capacity.
+            max_buffered_inputs: The maximum number of in-flight requests that
+                are allowed to be buffered. Before submitting more requests,
```

Comment:
hmm this is not the max number of inflight requests but rather num of buffers per channel. For example, if there are two channels, then the max in-flight requests is at least 2 * channel_num_buffers

Reply:
actually I'd like to keep the semantic of max_in_flight_requests. What about

- we change it to num_in_flight_requests
- We use num_in_flight_requests * 2 as the num buffer size only for shm channel?

Reply:
It would be even better if we could avoid adding the new config and just use `self._max_buffered_results`. They are trying to accomplish the same thing, I believe, just whether we buffer on the input or output.

Reply:
Also, it's okay if the max buffered results is conservative, i.e. the DAG has more capacity than what is requested through max buffered results. This can always be improved later.

For now, I would say the best is to determine the number of buffers per channel to use based on the existing `max_buffered_results` config (which we can rename to something like `max_inflight_executions`) and additionally add a "global memory limit" for all buffers' sizes, in case the user puts something very high like 1M max in-flight executions.

Reply:
Sounds great! How about just filing issues for now?
- global memory limit for buffers
- dynamically allocating additional buffers